docs/build/guides/account-linking-with-dapper.md

```diff
@@ -0,0 +1,752 @@
+
+```tsx
+<body suppressHydrationWarning={true}>
+```
+
+Your code should be:
+
+```tsx
+// layout.tsx
+'use client';
+import "./globals.css";
+import flowJSON from '../flow.json'
+import * as fcl from "@onflow/fcl";
+
+import AuthProvider from "./providers/AuthProvider";
+
+fcl.config({
+  "discovery.wallet": "https://fcl-discovery.onflow.org/authn",
+  'accessNode.api': process.env.NEXT_PUBLIC_ACCESS_NODE_API,
+  'flow.network': process.env.NEXT_PUBLIC_FLOW_NETWORK,
+  'walletconnect.projectId': process.env.NEXT_PUBLIC_WALLETCONNECT_ID
+}).load({ flowJSON });
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <html lang="en">
+      <body suppressHydrationWarning={true}>
+        <AuthProvider>
+          {children}
+        </AuthProvider>
+      </body>
+    </html>
+  );
+}
+```
+
+### Add the Connect Button
+
+Open `page.tsx` and clean up the demo code leaving only the `<main>` block:
+
+```tsx
+import Image from "next/image";
+
+export default function Home() {
+  return (
+    <div className="grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]">
+      <main className="flex flex-col gap-8 row-start-2 items-center sm:items-start">
+        <div>TODO</div>
+      </main>
+    </div>
+  );
+}
+```
+
+Add a `'use client';` directive, import the `useAuth` hook and instantiate it in the `Home` function:
+
+```tsx
+'use client';
+import { useAuth } from "./providers/AuthProvider";
+```
+
+```tsx
+const { user, loggedIn, logIn, logOut } = useAuth();
+```
+
+Then add a button in the `<main>` to handle logging in or out:
+
+```tsx
+<main className="flex flex-col gap-8 row-start-2 items-center sm:items-start">
+  <div>Welcome</div>
+  <button
+    onClick={loggedIn ? logOut : logIn}
+    className="px-6 py-2 text-white bg-green-600 hover:bg-green-700 rounded-lg shadow-md transition duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-green-500 sm:ml-auto"
+  >
+    {loggedIn ? "Log Out" : "Log In"}
+  </button>
+</main>
+```
+
+## Testing Pass
+
+Run the app:
+
+```zsh
+yarn dev
+```
+
+You'll see your `Log In` button in the middle of the window.
+
+![Welcome](welcome.png)
+
+Click the button and log in with your Flow wallet.
+
+![Flow Wallet](flow-wallet.png)
+
+## Account Linking
+
+Now that your app is set up, you can make use of [Account Linking] to to pull your NFTs from your Dapper Wallet, through your Flow Wallet, and into the app.
+
+### Set Up Account Linking
+
+If you haven't yet, you'll need to [link your Dapper Wallet] to your Flow Wallet.
+
+:::warning
+
+Dapper requires that you complete KYC before you can use Account Linking.  While this may frustrate some members of the community, it makes is much easier for app developers to design onboarding rewards and bonuses that are less farmable.
```

Comment:
```suggestion
Dapper requires that you complete KYC before you can use Account Linking.  While this may frustrate some members of the community, it makes it much easier for app developers to design onboarding rewards and bonuses that are less farmable.
```